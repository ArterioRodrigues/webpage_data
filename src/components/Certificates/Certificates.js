import React from 'react';
import certificatesData from '../../data/certificates.json';

const Certificates = () => (
  <div className="container">
    <div className="certificates">
      <h2 className="container__title">certifications</h2>
      <h3 className="container__subtitle">Certificates that I have collected over time</h3>
      <ul>
        {certificatesData.map(certificate => (
          <div className="container__list" key={`${certificate.name}-div`}>
            <li key={certificate.url} className="certificates__entry">
              <div className="certificate__name">{certificate.name}</div>
              <div className="certificate__homepage">
                <a
                  href={certificate.url}
                  target="_blank"
                  rel="noopener noreferrer"
                  aria-label={`${certificate.name} Link`}
                >
                  View certificate
                </a>
              </div>
            </li>
          </div>
        ))}
      </ul>
    </div>
  </div>
);

export default Certificates;
