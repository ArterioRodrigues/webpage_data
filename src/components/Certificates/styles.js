/**
 * Certificates component styles.
 */
import Color from 'color';
import { sSize, mSize, textWhite, green, midBlue, red, sansFont } from '../../theme';

export default () => ({
  entry: {
    marginBottom: '30px',
  },

  name: {
    fontSize: mSize,
    fontWeight: 500,
    '&>a': {
      textTransform: 'uppercase',
    },
  },
  subsection: {
    marginTop: '10px',
    marginBottom: '30px',
  },
  subsection_entry: {
    listStyle: 'circle',
    marginBottom: '10px',
  },
  progress: {
    borderRadius: '6px',
    color: textWhite,
    fontFamily: sansFont,
    fontWeight: 500,
    fontSize: sSize,
    marginLeft: '10px',
    padding: '2px 4px',
  },
  progress_high: {
    background: Color(green)
      .darken(0.25)
      .string(),
  },

  progress_mid: {
    background: Color(midBlue)
      .darken(0.25)
      .string(),
  },

  progress_low: {
    background: Color(red)
      .darken(0.2)
      .string(),
  },
});
