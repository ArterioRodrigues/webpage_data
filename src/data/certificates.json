{
  "sectionTitle": "certifications",
  "sectionSubtitle": "Certificates that I have collected over time",
  "content": [
    {
      "name": "Responsive Web Design Certification",
      "url": "https://www.freecodecamp.org/certification/caglarturali/responsive-web-design"
    },
    {
      "name": "Introduction to Game Development",
      "url": "https://www.coursera.org/account/accomplishments/certificate/GKDBNAGKPD6M"
    },
    {
      "name": "Introduction to Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/UC3L666SD5TG"
    },
    {
      "name": "3D Models for Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/4YHK3LYWMJPV"
    },
    {
      "name": "The Art of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/DNVYLQQL3WVF"
    },
    {
      "name": "The Technology of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/VZDUNEC24XER"
    },
    {
      "name": "Pro Tools Basics",
      "url": "https://www.coursera.org/account/accomplishments/certificate/YW4QY26UHC5F"
    }
  ]
}
