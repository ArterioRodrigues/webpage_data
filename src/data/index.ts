export * from './certificates';
export * from './contact';
export * from './education';
export * from './greeter';
export * from './metadata';
export * from './projects';
export * from './skills';
