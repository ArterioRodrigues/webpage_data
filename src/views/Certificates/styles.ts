/**
 * Certificates component styles.
 */
import { Styles } from 'react-jss';
import Color from 'color';
import {
  sSize,
  mSize,
  textWhite,
  green,
  midBlue,
  red,
  sansFont,
} from '../../theme';

export default {
  entry: {
    marginBottom: '30px',
  },
  name: {
    fontSize: mSize,
    fontWeight: 500,
    '&>a': {
      textTransform: 'uppercase',
    },
  },
  subsection: {
    marginTop: '10px',
    marginBottom: '30px',
  },
  subsectionEntry: {
    listStyle: 'circle',
    marginBottom: '10px',
  },
  progress: {
    borderRadius: '6px',
    color: textWhite,
    fontFamily: sansFont,
    fontWeight: 500,
    fontSize: sSize,
    marginLeft: '10px',
    padding: '2px 4px',
  },
  progressHigh: {
    background: Color(green).darken(0.25).string(),
  },

  progressMid: {
    background: Color(midBlue).darken(0.25).string(),
  },

  progressLow: {
    background: Color(red).darken(0.2).string(),
  },
} as Styles;
