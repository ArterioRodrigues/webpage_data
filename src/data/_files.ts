import { NavLink } from '../models';
import Greeter from '../views/Greeter';
import Skills from '../views/Skills';
import Education from '../views/Education';
import Projects from '../views/Projects';
import Certificates from '../views/Certificates';
import Contact from '../views/Contact';
import { certificates, contact, education, projects, skills } from '.';

export const files: NavLink[] = [
  {
    icon: ['fab', 'html5'],
    color: '#e44d26',
    url: '/',
    name: 'index.html',
    route: {
      component: Greeter,
      data: contact,
    },
  },
  {
    icon: ['fab', 'js'],
    color: '#ffca28',
    url: '/skills',
    name: 'skills.js',
    route: {
      component: Skills,
      data: skills,
    },
  },
  {
    icon: 'university',
    color: '#42a5f5',
    url: '/education',
    name: '.educationrc',
    route: {
      component: Education,
      data: education,
    },
  },
  {
    icon: 'sliders-h',
    color: '#98c379',
    url: '/projects',
    name: 'projects.config',
    route: {
      component: Projects,
      data: projects,
    },
  },
  {
    icon: ['fab', 'sass'],
    color: '#e13e76',
    url: '/certifications',
    name: 'certifications.sass',
    route: {
      component: Certificates,
      data: certificates,
    },
  },
  {
    icon: ['fab', 'markdown'],
    color: '#abb2bf',
    url: '/contact',
    name: 'contact.md',
    route: {
      component: Contact,
      data: contact,
    },
  },
];
