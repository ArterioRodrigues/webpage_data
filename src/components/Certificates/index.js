/**
 * Certificates component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import clsx from 'clsx';
import Container from '../Container';
import certificatesData from '../../data/certificates.json';
import styles from './styles';

const useStyles = createUseStyles(styles);

const Certificates = () => {
  const classes = useStyles();

  const listSubItems = (subItems) => (
    <ul className={classes.subsection}>
      {subItems.map((subCert) => (
        <li key={`${subCert.name}-div`} className={classes.subsection_entry}>
          <a
            href={subCert.url}
            target="_blank"
            rel="noopener noreferrer"
            aria-label={`${subCert.name} Link`}
          >
            {subCert.name}
          </a>
        </li>
      ))}
    </ul>
  );

  const buildProgressText = (total, completed) => {
    const percentage = Math.round((completed / total) * 100);

    let classToUse = '';
    if (percentage >= 75) {
      classToUse = 'progress_high';
    } else if (percentage >= 50) {
      classToUse = 'progress_mid';
    } else {
      classToUse = 'progress_low';
    }

    return (
      <span
        className={clsx(classes.progress, classes[classToUse])}
      >{`${percentage}%`}</span>
    );
  };

  return (
    <Container
      title={certificatesData.sectionTitle}
      subtitle={certificatesData.sectionSubtitle}
    >
      <ul>
        {certificatesData.content.map((certificate) => (
          <div className="container__list" key={`${certificate.name}-div`}>
            <li key={certificate.url} className="certificates__entry">
              <div className="certificate__name">
                <a
                  href={certificate.url}
                  target="_blank"
                  rel="noopener noreferrer"
                  aria-label={`${certificate.name} Link`}
                >
                  {certificate.name}
                  {buildProgressText(
                    certificate.total_courses ? certificate.total_courses : 1,
                    certificate.sub_items ? certificate.sub_items.length : 1,
                  )}
                </a>
                {certificate.sub_items && listSubItems(certificate.sub_items)}
              </div>
            </li>
          </div>
        ))}
      </ul>
    </Container>
  );
};

export default Certificates;
