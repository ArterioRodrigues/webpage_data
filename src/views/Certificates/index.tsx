/**
 * Certificates component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import clsx from 'clsx';
import Container from '../../components/Container';
import { Certificate, Linkable, BaseProps } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export type CertificatesProps = BaseProps<Certificate<Linkable>>;

const Certificates: React.FC<CertificatesProps> = ({ data, fileName }) => {
  const classes = useStyles();

  const listSubItems = (subItems: Linkable[]): React.ReactNode => (
    <ul className={classes.subsection}>
      {subItems.map(({ name, url }) => (
        <li key={`${name}-div`} className={classes.subsection_entry}>
          <a
            href={url}
            target="_blank"
            rel="noopener noreferrer"
            aria-label={`${name} Link`}
          >
            {name}
          </a>
        </li>
      ))}
    </ul>
  );

  const buildProgressText = (
    total: number,
    completed: number,
  ): React.ReactNode => {
    const percentage = Math.round((completed / total) * 100);

    let classToUse = '';
    if (percentage >= 75) {
      classToUse = 'progressHigh';
    } else if (percentage >= 50) {
      classToUse = 'progressMid';
    } else {
      classToUse = 'progressLow';
    }

    return (
      <span
        className={clsx(classes.progress, classes[classToUse])}
      >{`${percentage}%`}</span>
    );
  };

  return (
    <Container
      title={data.name}
      subtitle={data.desc}
      seo={{ title: fileName, description: data.desc }}
    >
      <ul>
        {data.content.map(({ name, url, totalCourses, earned }) => (
          <div key={`${name}-div`}>
            <li key={url} className={classes.entry}>
              <div className={classes.name}>
                <a
                  href={url}
                  target="_blank"
                  rel="noopener noreferrer"
                  aria-label={`${name} Link`}
                >
                  {name}
                  {buildProgressText(totalCourses, earned.length)}
                </a>
                {earned && listSubItems(earned)}
              </div>
            </li>
          </div>
        ))}
      </ul>
    </Container>
  );
};

export default Certificates;
