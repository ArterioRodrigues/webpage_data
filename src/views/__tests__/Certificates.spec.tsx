import React from 'react';
import { render } from '@testing-library/react';
import Certificates from '../Certificates';
import { certificates } from '../../data';

describe('<Certificates />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <Certificates data={certificates} fileName="certificates.test" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
