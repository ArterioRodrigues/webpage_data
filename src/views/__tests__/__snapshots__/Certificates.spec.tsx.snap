// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Certificates /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root-0-2-9"
  >
    <h2
      class="title-0-2-10"
    >
      certifications
    </h2>
    <h3
      class="subtitle-0-2-11"
    >
      Certificates that I have earned over time
    </h3>
    <div
      class="content-0-2-12"
    >
      <ul>
        <div>
          <li
            class="entry-0-2-1"
          >
            <div
              class="name-0-2-2"
            >
              <a
                aria-label="Full-Stack Web Development with React Specialization Link"
                href="https://www.coursera.org/account/accomplishments/specialization/certificate/XGPU5C68VAEE"
                rel="noopener noreferrer"
                target="_blank"
              >
                Full-Stack Web Development with React Specialization
                <span
                  class="progress-0-2-5 progressHigh-0-2-6"
                >
                  100%
                </span>
              </a>
              <ul
                class="subsection-0-2-3"
              >
                <li>
                  <a
                    aria-label="Front-End Web UI Frameworks and Tools: Bootstrap 4 Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/TQ2HAVM6JJ73"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Front-End Web UI Frameworks and Tools: Bootstrap 4
                  </a>
                </li>
                <li>
                  <a
                    aria-label="Front-End Web Development with React Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/QJPWRV28SCV3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Front-End Web Development with React
                  </a>
                </li>
                <li>
                  <a
                    aria-label="Multiplatform Mobile App Development with React Native Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/GNX5WEXJWSGY"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Multiplatform Mobile App Development with React Native
                  </a>
                </li>
                <li>
                  <a
                    aria-label="Server-Side Development with NodeJS, Express and MongoDB Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/XGMAYMHEXDZW"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Server-Side Development with NodeJS, Express and MongoDB
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </div>
        <div>
          <li
            class="entry-0-2-1"
          >
            <div
              class="name-0-2-2"
            >
              <a
                aria-label="freeCodeCamp | The Full Stack Development Certificate Link"
                href="https://guide.freecodecamp.org/certifications/"
                rel="noopener noreferrer"
                target="_blank"
              >
                freeCodeCamp | The Full Stack Development Certificate
                <span
                  class="progress-0-2-5 progressLow-0-2-8"
                >
                  33%
                </span>
              </a>
              <ul
                class="subsection-0-2-3"
              >
                <li>
                  <a
                    aria-label="Responsive Web Design Certification Link"
                    href="https://www.freecodecamp.org/certification/caglarturali/responsive-web-design"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Responsive Web Design Certification
                  </a>
                </li>
                <li>
                  <a
                    aria-label="JavaScript Algorithms and Data Structures Certification Link"
                    href="https://www.freecodecamp.org/certification/caglarturali/javascript-algorithms-and-data-structures"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    JavaScript Algorithms and Data Structures Certification
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </div>
        <div>
          <li
            class="entry-0-2-1"
          >
            <div
              class="name-0-2-2"
            >
              <a
                aria-label="Game Design and Development Specialization Link"
                href="https://www.coursera.org/specializations/game-development"
                rel="noopener noreferrer"
                target="_blank"
              >
                Game Design and Development Specialization
                <span
                  class="progress-0-2-5 progressLow-0-2-8"
                >
                  20%
                </span>
              </a>
              <ul
                class="subsection-0-2-3"
              >
                <li>
                  <a
                    aria-label="Introduction to Game Development Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/GKDBNAGKPD6M"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Introduction to Game Development
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </div>
        <div>
          <li
            class="entry-0-2-1"
          >
            <div
              class="name-0-2-2"
            >
              <a
                aria-label="Virtual Reality Specialization Link"
                href="https://www.coursera.org/specializations/virtual-reality"
                rel="noopener noreferrer"
                target="_blank"
              >
                Virtual Reality Specialization
                <span
                  class="progress-0-2-5 progressLow-0-2-8"
                >
                  40%
                </span>
              </a>
              <ul
                class="subsection-0-2-3"
              >
                <li>
                  <a
                    aria-label="Introduction to Virtual Reality Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/UC3L666SD5TG"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Introduction to Virtual Reality
                  </a>
                </li>
                <li>
                  <a
                    aria-label="3D Models for Virtual Reality Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/4YHK3LYWMJPV"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    3D Models for Virtual Reality
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </div>
        <div>
          <li
            class="entry-0-2-1"
          >
            <div
              class="name-0-2-2"
            >
              <a
                aria-label="Music Production Specialization Link"
                href="https://www.coursera.org/specializations/music-production"
                rel="noopener noreferrer"
                target="_blank"
              >
                Music Production Specialization
                <span
                  class="progress-0-2-5 progressHigh-0-2-6"
                >
                  75%
                </span>
              </a>
              <ul
                class="subsection-0-2-3"
              >
                <li>
                  <a
                    aria-label="The Art of Music Production Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/DNVYLQQL3WVF"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    The Art of Music Production
                  </a>
                </li>
                <li>
                  <a
                    aria-label="The Technology of Music Production Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/VZDUNEC24XER"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    The Technology of Music Production
                  </a>
                </li>
                <li>
                  <a
                    aria-label="Pro Tools Basics Link"
                    href="https://www.coursera.org/account/accomplishments/certificate/YW4QY26UHC5F"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Pro Tools Basics
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </div>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
